src/Components/ServiceScene/LogsPanelScene.tsx

```diff
@@ -20,28 +22,93 @@ import { reportAppInteraction, USER_EVENTS_ACTIONS, USER_EVENTS_PAGES } from '..
 import { CopyLinkButton } from './CopyLinkButton';
-import { getLogsPanelSortOrder, LogOptionsScene } from './LogOptionsScene';
+import { getLogsPanelSortOrderFromStore, LogOptionsScene } from './LogOptionsScene';
 import { LogsVolumePanel, logsVolumePanelKey } from './LogsVolumePanel';
 import { getPanelWrapperStyles, PanelMenu } from '../Panels/PanelMenu';
 import { ServiceScene } from './ServiceScene';
+import { Options } from '@grafana/schema/dist/esm/raw/composable/logs/panelcfg/x/LogsPanelCfg_types.gen';
+import { locationService } from '@grafana/runtime';
+import { narrowLogsSortOrder } from '../../services/narrowing';
+import { logger } from '../../services/logger';
+import { LogsSortOrder } from '@grafana/schema';
 
 interface LogsPanelSceneState extends SceneObjectState {
-  body?: VizPanel;
+  body?: VizPanel<Options>;
+  sortOrder?: LogsSortOrder;
+  wrapLogMessage?: boolean;
 }
 
 export class LogsPanelScene extends SceneObjectBase<LogsPanelSceneState> {
+  protected _urlSync = new SceneObjectUrlSyncConfig(this, {
+    keys: ['sortOrder', 'wrapLogMessage'],
+  });
+
   constructor(state: Partial<LogsPanelSceneState>) {
     super({
+      sortOrder: getLogsPanelSortOrderFromStore(),
+      wrapLogMessage: Boolean(getLogOption<boolean>('wrapLogMessage', false)),
       ...state,
     });
 
     this.addActivationHandler(this.onActivate.bind(this));
   }
 
+  private setStateFromUrl() {
+    const searchParams = new URLSearchParams(locationService.getLocation().search);
+
+    this.updateFromUrl({
+      sortOrder: searchParams.get('sortOrder'),
+      wrapLogMessage: searchParams.get('wrapLogMessage'),
+    });
+  }
+
+  getUrlState() {
+    return {
+      sortOrder: JSON.stringify(this.state.sortOrder),
+      wrapLogMessage: JSON.stringify(this.state.wrapLogMessage),
+    };
+  }
+
+  updateFromUrl(values: SceneObjectUrlValues) {
+    const stateUpdate: Partial<LogsPanelSceneState> = {};
+    try {
+      if (typeof values.sortOrder === 'string') {
+        const decodedSortOrder = narrowLogsSortOrder(JSON.parse(values.sortOrder));
+        if (decodedSortOrder) {
+          stateUpdate.sortOrder = decodedSortOrder;
+          this.setLogsVizOption({ sortOrder: decodedSortOrder });
+        }
+      }
+
+      if (typeof values.wrapLogMessage === 'string') {
+        const decodedWrapLogMessage = JSON.parse(values.wrapLogMessage);
+        if (typeof decodedWrapLogMessage === 'boolean') {
+          stateUpdate.wrapLogMessage = decodedWrapLogMessage;
+          this.setLogsVizOption({ wrapLogMessage: decodedWrapLogMessage });
+          this.setLogsVizOption({ prettifyLogMessage: decodedWrapLogMessage });
+        }
+      }
+    } catch (e) {
+      // URL Params can be manually changed and it will make JSON.parse() fail.
+      logger.error(e, { msg: 'LogOptionsScene: updateFromUrl unexpected error' });
+    }
+
+    if (Object.keys(stateUpdate).length) {
+      this.setState({ ...stateUpdate });
+    }
+  }
+
```

Comment:
Do we want to store in store if we update from URL? I navigated from a link, had `wrapping: true`, went to service selection, selected a new service, and `wrapping: false`.
Would be similar question for sort order.

Reply:
I would argue we only want to update local storage state when the user makes an action, e.g. if I'm viewing a link that looks better with line wrap off, I wouldn't expect viewing that link to change my settings next time I start a new investigation

Reply:
But this isn't a hill I'm trying to die on, I can forsee situations where either situation is more desirable 🤷

Reply:
Yea, same. Let's go with the current state.